ext/openssl/openssl.c

```diff
@@ -6536,9 +6545,9 @@ PHP_FUNCTION(openssl_sign)
 	pkey = php_openssl_pkey_from_zval(key, 0, "", 0, &free_pkey);
 	if (pkey == NULL) {
 		if (!EG(exception)) {
-			php_error_docref(NULL, E_WARNING, "Supplied key param cannot be coerced into a private key");
+			zend_argument_type_error(3, "cannot be coerced into a private key");
```

Comment:
Should I revert this?

Reply:
All key and cert errors should be reverted to warning otherwise it's going be too inconsistent. The same for algorithm checks. Basically the only one that can stay is just number conversions.